### ISSUE TITLE
Bundler/InsecureProtocolSource doesn't check Git links for https

### ISSUE THREAD

**Issue by tibbon** [rubocop-hq/rubocop#5971]
Perhaps this needs to be a new cop (I'd write it if so). In a Gemfile, if I use an insecure Git link, this cop doesn't flag it. For example: 

```
gem 'rubocop', git: 'http://github.com/rubocop-hq/rubocop.git'
```

This should raise an error, and ideally would autocorrect to:

```
gem 'rubocop', git: 'https://github.com/rubocop-hq/rubocop.git'
```

## RuboCop version

```
$ rubocop -V
0.56.0 (using Parser 2.5.1.0, running on ruby 2.3.7 x86_64-darwin17)
```

**Comment by mikegee** [comment]
👍 makes sense

Should the autocorrect part check for "github.com" (and other known sources)? I'm thinking there are folks out there pulling a repo insecurely because it isn't available securely.

**Comment by rrosenblum** [comment]
> Should the autocorrect part check for "github.com" (and other known sources)? I'm thinking there are folks out there pulling a repo insecurely because it isn't available securely.

What do you think of the cop accepting a list of hosts that don't support https? That way we can still register offenses for non standard hosts.

Don't forget that the `git@` protocol is also secure. `git@github.com:rubocop-hq/rubocop.git`

**Comment by tibbon** [comment]
Yea, I think having domain-exclusions as an option would be a good idea.

**Comment by koic** [comment]
Perhaps it is better to add the following line to Gemfile and tell it to use the `github` option.

```ruby
git_source(:github) {|repo_name| "https://github.com/#{repo_name}" }
```

AFAIK, In Rails and Bundler, this line will be included by default.

- https://github.com/rails/rails/blob/v5.2.0/railties/lib/rails/generators/rails/app/templates/Gemfile.tt#L2
- https://github.com/bundler/bundler/blob/v1.16.2/lib/bundler/templates/Gemfile#L5
